src/server/pkg/storage/gc/client.go

```diff
@@ -0,0 +1,265 @@
+package gc
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/lib/pq"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/chunk"
+)
+
+type Server interface {
+	Ctx() context.Context
```

Comment:
It's generally considered bad practice on Go to store contexts in structs since they're supposed to be passed in for each request. We do this in our `pachclient` although have plans to stop doing that. Not sure how integral to the design having this in the Server implementation is, I'm not totally convinced we should have this as a hard and fast rule, but it seems to make sense as a guideline.